src/kixi/comms/components/coreasync.clj

```diff
@@ -0,0 +1,140 @@
+(ns kixi.comms.components.coreasync
+  (:require [clojure.core.async :as async]
+            [com.stuartsierra.component :as component]
+            [kixi.comms :as comms]
+            [kixi.comms.messages :as msg]
+            [taoensso.timbre :as timbre :refer [debug info error]]))
+
+(defn sanitize-app-name
+  [profile s]
+  (str "kixi-comms-"
+       (name profile) "-app-"
+       (-> s
+           (clojure.string/replace #"\:" "")
+           (clojure.string/replace #"\/" "_"))))
+
+(defn attach-generic-processing-switch
+  [target-chan id->handle-msg-and-process-msg-atom]
+  (async/go-loop [msg (async/<! target-chan)]
+    (when msg
+      (when comms/*verbose-logging*
+        (info "Received msg from Core Async: " msg))
+      (doseq [{:keys [process-msg? handle-msg app-name]} (vals @id->handle-msg-and-process-msg-atom)]
+                                        ;should process in parrel
+        (if (process-msg? msg)
+          (try
+            (debug "# Forwarding last message to handler" app-name)
+            (handle-msg msg)
+            (catch Throwable e
+              (error e "Handler threw an exception:" app-name msg)))
+          (debug "# NOT forwarding last message to handler" app-name)))
+      (recur (async/<! target-chan)))))
+
+(defn put-msg
+  [ch msg]
+  (let [opts (last msg)
+        formatted (apply msg/format-message (conj (vec (butlast msg)) opts))]
```

Comment:
don't know if it worth abstracting this away in the formatting? (the whole handling of the msg)